sbi/inference/posteriors/direct_posterior.py

```diff
@@ -276,8 +279,9 @@ def sample(
 
         if sample_with_mcmc:
             samples = self._sample_posterior_mcmc(
-                x=x,
                 num_samples=num_samples,
+                x=x,
+                potential_fn_provider=PotentialFunctionProvider(),
```

Comment:
This is hard coded, thus it will never need to get (user) input arguments?
This is because the provided will get prior, net, x, and mcmc_method upon `__call__()`?

Reply:
Yes, exactly